lcviewernoqt/painters/opengl/sources/cacher.cpp

```diff
@@ -0,0 +1,331 @@
+    _path_distance+=d;
+  }
+  _current_vertices.push_back( glm::vec4(x,-y,z,_path_distance) );
+}
+
+void Cacher::appendVertexData()
+{
+  if(_current_vertices.size()>1)
+  {
+    if(_fill==true)
+    {
+      _cacher_vertex_data.insert( _cacher_vertex_data.end() , _current_vertices.begin() , _current_vertices.end() );
+      _jumps.push_back(_current_vertices.size());
+    }
+
+    else
+    {
+      if(_closed==false)
+      {
+        _cacher_vertex_data.push_back( *(_current_vertices.begin()+1)  );  // 2nd
+        _cacher_vertex_data.insert( _cacher_vertex_data.end() , _current_vertices.begin() , _current_vertices.end() );
+        _cacher_vertex_data.push_back( *(_current_vertices.rbegin()+1) );    // 2nd Last
+        _jumps.push_back(_current_vertices.size()+2);
+      }
+
+      else
+      {
+        _cacher_vertex_data.push_back( *(_current_vertices.rbegin()) );  // last
+        _cacher_vertex_data.insert( _cacher_vertex_data.end() , _current_vertices.begin() , _current_vertices.end() );
+        _cacher_vertex_data.push_back( *(_current_vertices.begin())  );    // 1st
+        _cacher_vertex_data.push_back( *(_current_vertices.begin()+1)  );  // 2nd
+        _jumps.push_back(_current_vertices.size()+3);
+      }
+    }
+  }
+
+  else if(_current_vertices.size()==1)
+  {
+    _cacher_vertex_data.insert( _cacher_vertex_data.end() , _current_vertices.begin() , _current_vertices.end() );
+    _jumps.push_back(_current_vertices.size());
+  }
+
+  _current_vertices.clear();
+}
+
+void Cacher::jump()
+{
+    appendVertexData();
+    _closed=false;
+    _path_distance=0.0f;
+}
+
+void Cacher::closeLoop()
+{
+  _closed=true;
+}
+
+void Cacher::clearData()
+{
+  _closed=false;
+  _fill=false;
+  _path_distance=0.0f;
+  _cacher_vertex_data.clear();
+  _current_vertices.clear();
+  _jumps.clear();
+}
+
+//---------------------------------------------------------------
+void Cacher::addDataToCurrentEntity()
+{
+  appendVertexData();
+
+  _current_gl_entity->loadVertexData(&_cacher_vertex_data[0].x , _cacher_vertex_data.size()*(4*sizeof(float)) , _jumps );
+  _current_gl_entity->setModelMatrix(_model);          
+        
+  _current_gl_entity->setLineWidth(_cacher_line_width);                        // ALERT:
+  _current_gl_entity->setDashes(_cacher_dashes_data,_cacher_dashes_size,_cacher_dashes_sum);   // THIS Order
+  _current_gl_entity->setFillMode(_fill);                               // Is Fixed!!!
+  _current_gl_entity->setType(_shaders);
+  _current_gl_entity->setFont(_fonts,_font_style);
+  _current_gl_entity->addTextData(_cacher_vertex_data[0], _text_value, _text_height, _no_text_magnify);  
+}
+
+void Cacher::selectFill()
+{
+  _fill=true;  
+}
+
+void Cacher::selectColor(float R,float G,float B,float A)
+{
+  // Till now dont need to be cached
+}
+
+void Cacher::selectLineWidth(float width)
+{
+  _cacher_line_width=width;
+}
+
+void Cacher::selectDashes(const double* dashes, const int num_dashes, double offset, bool scaled)
```

Comment:
Exact code duplication with `Renderer::selectDashes(const double*, const int, double, bool)`
Renaming variables to make CodeFactor check passes doesn't remove the code duplication

Reply:
Done!!